yt/data_objects/index_subobjects/stretched_grid.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+
+from yt.geometry.selection_routines import convert_mask_to_indices
+
+from .grid_patch import AMRGridPatch
+
+
+class StretchedGrid(AMRGridPatch):
+    def __init__(self, id, cell_widths, filename=None, index=None):
+        self.cell_widths = [np.array(_) for _ in cell_widths]
+        super().__init__(id, filename, index)
+
+    def _check_consistency(self):
+        computed_right_edge = self.LeftEdge + [_.sum() for _ in self.cell_widths]
+        assert (computed_right_edge == self.RightEdge).all()
+
+    def _get_selector_mask(self, selector):
+        if self._cache_mask and hash(selector) == self._last_selector_id:
+            mask = self._last_mask
+        else:
+            mask = selector.fill_mask(self)
+            if self._cache_mask:
+                self._last_mask = mask
+            self._last_selector_id = hash(selector)
+            if mask is None:
+                self._last_count = 0
+            else:
+                self._last_count = mask.sum()
+        return mask
+
+    def select_fwidth(self, dobj):
+        mask = self._get_selector_mask(dobj.selector)
+        if mask is None:
+            return np.empty((0, 3), dtype="float64")
+        indices = convert_mask_to_indices(mask, self._last_count)
+
+        coords = np.array(
+            [
+                self.cell_widths[0][indices[:, 0]],
+                self.cell_widths[1][indices[:, 1]],
+                self.cell_widths[2][indices[:, 2]],
+            ]
+        ).T
+        return coords
+
+    def select_fcoords(self, dobj):
+        mask = self._get_selector_mask(dobj.selector)
+        if mask is None:
+            return np.empty((0, 3), dtype="float64")
+        cell_centers = [
+            self.LeftEdge[i]
+            + np.add.accumulate(self.cell_widths[i])
```

Comment:
There's a bug here because `self.LeftEdge` is presumably a `unyt_array` with length dimensionality, while `np.add.accumulate` currently is *not* supported by unyt (it doesn't preserve units). If you mean to get unitless arrays here this could be a fix, although we should also double check that `LeftEdge` and `cell_widths` are stored using the same unit.
```suggestion
            self.LeftEdge[I].d
            + np.add.accumulate(self.cell_widths[i])
```

Reply:
I further report this is the *only* blocking bug for yt_idefix https://github.com/neutrinoceros/yt_idefix/pull/120
🎉

Reply:
After further inspection I *think* it's safe to assume both cell_widths and LeftEdge are provided in `"code_length"` units, so the minimal suggestion _should_ be self-sufficient

Reply:
I actually don't know why this works for me -- do you?

Reply:
(i.e., why didn't I see this bug?)